presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraTable.java

```diff
@@ -49,6 +51,31 @@ public CassandraTableHandle getTableHandle()
                 .collect(toList());
     }
 
+    public List<CassandraColumnHandle> getClusteringKeyColumns()
+    {
+        return columns.stream()
+                .filter(CassandraColumnHandle::isClusteringKey)
+                .collect(toList());
+    }
+
+    public boolean isLastClusteringKeyColumn(CassandraColumnHandle columnHandle)
+    {
+        List<CassandraColumnHandle> clusteringKeyColumns = getClusteringKeyColumns();
+        return (clusteringKeyColumns.get(clusteringKeyColumns.size() - 1).equals(columnHandle));
+    }
+
+    public boolean isAPrefix(Set<CassandraColumnHandle> columnHandles)
```

Comment:
I don't think you would need this method once you refactor `unenforcedPredicates`